test/controllers/trusts.involved.controller.spec.ts

```diff
@@ -200,14 +212,149 @@ describe('Trust Involved controller', () => {
+      expect(mapTrustWhoIsInvolvedToPage).toBeCalledWith(mockAppData, trustId, false);
+
+      expect(mockRes.render).toBeCalledTimes(1);
+      expect(mockRes.render).toBeCalledWith(
+        TRUST_INVOLVED_PAGE,
+        expect.objectContaining({
+          pageData: expect.objectContaining({
+            trustData: mockTrustData,
+            ...mockInvolvedData,
+          }),
+          errors: {
+            errorList: [
+              {
+                href: '#typeOfTrustee',
+                text: ErrorMessages.TRUST_INVOLVED_INVALID,
+              },
+            ],
+            typeOfTrustee: {
+              text: ErrorMessages.TRUST_INVOLVED_INVALID,
+            },
+          },
+          formData: mockReq.body,
+        }),
+      );
+    });
+
+    test('catch error when post data from page', async () => {
+      mockReq.body = {
+        id: 'dummyId',
+        typeOfTrustee: 'dummyTrusteeType',
+        noMoreToAdd: 'add',
+      };
+      const error = new Error(ANY_MESSAGE_ERROR);
+      (mockRes.redirect as jest.Mock).mockImplementationOnce(() => {
+        throw error;
+      });
+
+      await post(mockReq, mockRes, mockNext);
+
+      expect(mockNext).toBeCalledTimes(1);
+      expect(mockNext).toBeCalledWith(error);
+    });
+  });
+
+  describe('POST with url params unit tests', () => {
+    // TODO - get working when trust-involved no more to add button ticket is worked on
+    // test('no more to add button pushed', async () => {
+    //   mockReq.body = {
+    //     noMoreToAdd: 'noMoreToAdd',
+    //   };
+
+    //   mockIsActiveFeature.mockReturnValueOnce(true);
+
+    //   await post(mockReq, mockRes, mockNext);
+
+    //   expect(mockRes.redirect).toBeCalledTimes(1);
+    //   expect(mockRes.redirect).toBeCalledWith(`${TRUST_ENTRY_URL + ADD_TRUST_URL}`);
+    // });
+
+    const dpPostTrustee = [
+      [
+        TrusteeType.HISTORICAL,
+        TRUST_HISTORICAL_BENEFICIAL_OWNER_URL,
+      ],
+      [
+        TrusteeType.INDIVIDUAL,
+        TRUST_INDIVIDUAL_BENEFICIAL_OWNER_URL,
+      ],
+      [
+        TrusteeType.LEGAL_ENTITY,
+        TRUST_LEGAL_ENTITY_BENEFICIAL_OWNER_URL,
+      ],
+      [
+        'unknown',
+        '',
+      ],
+    ];
+
+    test.each(dpPostTrustee)(
+      'success push with %p type',
+      async (typeOfTrustee: string, expectedUrl: string) => {
+        mockReq.body = {
+          typeOfTrustee,
+        };
+
+        (validationResult as any as jest.Mock).mockImplementationOnce(() => ({
+          isEmpty: jest.fn().mockReturnValue(true),
+        }));
+
+        mockIsActiveFeature.mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_REDIS_REMOVAL
+
+        await post(mockReq, mockRes, mockNext);
+
+        expect(mockRes.redirect).toBeCalledTimes(1);
+        expect(mockRes.redirect).toBeCalledWith(`${MOCKED_URL}/${trustId}${expectedUrl}`);
+      },
+    );
+
+    // TODO - get working when trust-involved validation ticket is worked on
```

Comment:
Is this TODO comment still required as test is now uncommented? (presume it passes...)

Reply:
i missed it, it can go